xarray/core/dataset.py

```diff
@@ -2270,6 +2270,8 @@ def _isel_fancy(
                 }
                 if var_indexers:
                     new_var = var.isel(indexers=var_indexers)
+                    if drop and new_var.ndim == 0:
```

Comment:
Could we link back to this issue (or explain inline, but linking back probably easier)?